### PR TITLE
Generate XCKernel from LibXC string

### PR DESCRIPTION
**Pull request by dmejiar**
- Allows to define a functional using a string like `"xc_gga_x_ncap"`  when LibXC is used as backend.
- Updates LibXC version to 6.1.0. 
Note that LibXC 5.0 was buggy (see [ChangeLog.md](https://gitlab.com/libxc/libxc/-/blob/b716b37a330b5d87e49eff9dec8ed2ca39fcd69e/ChangeLog.md)) and should not be taken as reference.

### CODE REVIEW

**Comment by dmejiar** [comment]
@wavefunction91 I have updated the reference test data. Seems that only three values were off.  PR #8 should work with the current master branch of LibXC, but I will double check that

**Comment by wavefunction91** [comment]
Thanks for the corrections. However, the main problem was not in the reference values, but in the comparison of Libxc/Bulitin backends. Recall, the fact that `ExchCXX` can act as a `Libxc` wrapper is convenient, but not a necessity - all changes must be propagated to the Builtin backend to allow for their evaluation on GPUs (which is primary purpose for this project).

I tried to regenerate the Builtin interfaces from Libxc 6.0.0, but the've changed there backend function structure, so it will take me a bit to update my scripts.

I've enabled this PR to run through the CI pipeline, the errors should be apparent now.

**Comment by wavefunction91** [comment]
@dmejiar please merge with current `master` for CI pipeline fixes

**Comment by dmejiar** [comment]
I started updating the built-in kernels, but there is another issue that can lead to some test failures. Libxc corrects some unphysical `sigma` values before evaluating the functional. These corrections are not present in ExchCXX.

The last three sigma values (1.3, 1.5, 1.5) in [Line 56](
https://github.com/wavefunction91/ExchCXX/blob/e255fc511bdfc106da537031882ff0412df5c9f0/test/reference_values.cxx#L56 ) will lead to an unphysical situation. Libxc will instead use `sigma_aa = 1.3`, `sigma_ab = 1.4`, and `sigma_bb = 1.5`.

Do you want to have the same checks on `sigma` as in LibXC? If you do, then I can leave the values of the `sigma_polarized` as they are. But if you do not want those checks, then I will change the values to (1.3, 1.4, 1.5) so there is really a one-to-one comparison

**Comment by wavefunction91** [comment]
@dmejiar Thanks for taking a look. I think we ideally want both - the checks are clearly necessary, but we also want to have sane values to test with. Does LibXC error out with unphysical `sigma`?

**Comment by dmejiar** [comment]
No, Libxc will correct the unphysical value

```
      s_ave = 0.5*(my_sigma[0] + my_sigma[2]);
      /* | grad n |^2 = |grad n_up + grad n_down|^2 > 0 */
      my_sigma[1] = (my_sigma[1] >= -s_ave ? my_sigma[1] : -s_ave);
      /* Since |grad n_up - grad n_down|^2 > 0 we also have */
      my_sigma[1] = (my_sigma[1] <= +s_ave ? my_sigma[1] : +s_ave);
```

**Comment by dmejiar** [comment]
@wavefunction91 

Tests already pass with 4c92dda [see here](https://github.com/dmejiar/ExchCXX/actions/runs/4835507291/jobs/8617872937).

Commit 044b12a updates the generation script to handle the new structure in Libxc 6

**Comment by ajaypanyala** [comment]
@dmejiar Are the changes in [here](https://github.com/dmejiar/ExchCXX/commit/1956a7f39fcf79c6534d98b9aae259d11f100b7d) (reg. `set_ext_params`) also supposed to go into this PR ?

**Comment by dmejiar** [comment]
@ajaypanyala Only a handful of specialized functionals need external parameters to be set, so they are not really necessary, but I can include them as well. What do you think?

**Comment by wavefunction91** [comment]
I'm fine exposing the functionality, but none of the builtin kernels support this functionality. Recalling that Libxc-wrapping is a convenient (but extra) feature for this project, it should be expected that all publicly exposed APIs should be useful to any of the implemented functions. 

That being said, if you want to expose it, that's fine, just make sure we throw and error / warning if the function is called from the builtin backend.

**Comment by dmejiar** [comment]
@ajaypanyala The `qed-hf` application does need to set external parameters. If I remember correctly, it might be possible to pass those parameters to Libxc by writing a file inside the running directory. Let me check if this is indeed possible. In this way, we would not need to expose `set_ext_params`

File: CMakeLists.txt [hunk 1]
```diff
@@ -56,23 +56,23 @@ endif()
...
   endif()
 
 else()
 
   FetchContent_Declare(
     libxc
-    GIT_REPOSITORY https://gitlab.com/eduard1/libxc.git
-    GIT_TAG        53883f416d0f421e4e856171d72c81fb331d8a57  # v5.0.0 + pr324 + pr351
+    GIT_REPOSITORY https://gitlab.com/libxc/libxc.git
+    GIT_TAG 6.1.0  # v6.1.0
```
Comment (wavefunction91) [inline CMakeLists.txt:73]: Do you want to track 6.0.0 or 6.1.0 here? `find_package` check is for 6.0.0.

**Review by wavefunction91 — DISMISSED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (dmejiar) [reply CMakeLists.txt:73]: I have updated the `find_package` check to 6.1.0

**Review by wavefunction91 — APPROVED** [review]

**Pull request merged**